test/components/DayPicker_spec.jsx

```diff
@@ -832,6 +833,17 @@ describe('DayPicker', () => {
     });
   });
 
+  describe('#getWeekHeaders', () => {
+    it('returns unmutated weekday headers for currentMonth in a future', () => {
+      const INITIAL_MONTH = moment().add(2, 'Months').week(3).weekday(3);
+      var dayPicker = new PureDayPicker(PureDayPicker.defaultProps);
+      dayPicker.state.currentMonth = moment(INITIAL_MONTH);
+      const state = cloneDeep(dayPicker.state);
+      expect(dayPicker.getWeekHeaders()).to.be.eql(INITIAL_MONTH.localeData().weekdaysMin());
+      expect(dayPicker.state).to.eql(state);
```

Comment:
if you also want to ensure they're not mutated, add this:
```suggestion
      expect(dayPicker.state).not.to.equal(state);
      expect(dayPicker.state).to.eql(state);
```